netlib/http2/frame.py

```diff
@@ -20,18 +27,53 @@ class Frame(object):
     FLAG_PADDED = 0x8
     FLAG_PRIORITY = 0x20
 
-    def __init__(self, length, flags, stream_id):
+    def __init__(
+            self,
+            state=None,
+            length=0,
+            flags=FLAG_NO_FLAGS,
+            stream_id=0x0):
         valid_flags = reduce(lambda x, y: x | y, self.VALID_FLAGS, 0x0)
         if flags | valid_flags != valid_flags:
             raise ValueError('invalid flags detected.')
 
+        if state is None:
+            from . import HTTP2Protocol
```

Comment:
We should avoid inline import statements as much as possible (hello, PyInstaller). Any reason why we can't place that at the top?